<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>335</width>
    <height>432</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Morse Code Generator</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>&amp;Words per Minute (WPM)</string>
           </property>
           <property name="buddy">
            <cstring>wpmSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QDoubleSpinBox" name="wpmSpinBox">
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>300.000000000000000</double>
           </property>
           <property name="value">
            <double>5.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Di&amp;t factor</string>
           </property>
           <property name="buddy">
            <cstring>ditFactorSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QDoubleSpinBox" name="ditFactorSpinBox">
           <property name="toolTip">
            <string>Enlarge/reduce length of dots (dits) by this factor.</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>10000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Da&amp;h factor</string>
           </property>
           <property name="buddy">
            <cstring>dahFactorSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QDoubleSpinBox" name="dahFactorSpinBox">
           <property name="toolTip">
            <string>Enlarge/reduce length of dahs by this factor.</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>10000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>&amp;Intra-character factor</string>
           </property>
           <property name="buddy">
            <cstring>intraFactorSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="intraFactorSpinBox">
           <property name="toolTip">
            <string>Enlarge/reduce silence between dits/dahs inside a character by this factor.</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>10000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>&amp;Character factor</string>
           </property>
           <property name="buddy">
            <cstring>charFactorSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QDoubleSpinBox" name="charFactorSpinBox">
           <property name="toolTip">
            <string>Enlarge/reduce silence between characters by this factor.</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>10000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>&amp;Word factor</string>
           </property>
           <property name="buddy">
            <cstring>wordFactorSpinBox</cstring>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QDoubleSpinBox" name="wordFactorSpinBox">
           <property name="toolTip">
            <string>Enlarge/reduce silence between words by this factor.</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
           <property name="decimals">
            <number>2</number>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>10000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>resulting WPM</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QLabel" name="wpmLabel">
           <property name="font">
            <font>
             <weight>50</weight>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>?</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QLabel" name="morseChar">
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QLabel" name="morseSymbol">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="alignment">
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>current character</string>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>current morse symbol</string>
           </property>
          </widget>
         </item>
         <item row="7" column="0">
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>82</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="maximum">
          <number>90</number>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="MorseScroller" name="scrollWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>15</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="morseStartButton">
           <property name="text">
            <string>Start</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="morseStopButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QCheckBox" name="loopCheckBox">
           <property name="text">
            <string>Loop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>335</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MorseScroller</class>
   <extends>QLabel</extends>
   <header>scroller.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>wpmSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>wpmChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>239</x>
     <y>79</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ditFactorSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>ditFactorChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dahFactorSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>dahFactorChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>134</y>
    </hint>
    <hint type="destinationlabel">
     <x>375</x>
     <y>144</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>intraFactorSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>intraFactorChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>160</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>169</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>charFactorSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>charFactorChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>192</y>
    </hint>
    <hint type="destinationlabel">
     <x>397</x>
     <y>202</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>wordFactorSpinBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>MainWindow</receiver>
   <slot>wordFactorChanged(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>236</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>morseStartButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>morseGeneratorStart()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>49</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>300</x>
     <y>277</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>morseStopButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>morseGeneratorStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>344</y>
    </hint>
    <hint type="destinationlabel">
     <x>289</x>
     <y>306</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>wpmChanged(double)</slot>
  <slot>ditFactorChanged(double)</slot>
  <slot>dahFactorChanged(double)</slot>
  <slot>intraFactorChanged(double)</slot>
  <slot>charFactorChanged(double)</slot>
  <slot>wordFactorChanged(double)</slot>
  <slot>morseGeneratorFinished()</slot>
  <slot>morseGeneratorStart()</slot>
  <slot>morseGeneratorStop()</slot>
 </slots>
</ui>
